src/Engine/EmitMSIL/CodeGenIL.cs

```diff
@@ -19,6 +19,29 @@ namespace EmitMSIL
 {
     public class CodeGenIL : IDisposable
     {
+        /// <summary>
+        /// Run modes. 
+        /// </summary>
+        public enum RunMode
+        {
+            /// <summary>
+            /// Compile graph and execute compiled assembly in memory.
+            /// </summary>
+            CompileAndExecute,
```

Comment:
do we really need two of these enums? Is the intention to split the internal and API layer?

Reply:
DynamoApplication, DynamoCore , and EmitMSIL projects need access to this enum and not all of them reference each other.